packages/edit-site/src/components/sidebar-dataviews/default-views.js

```diff
@@ -68,21 +69,59 @@ const DEFAULT_POST_BASE = {
 	layout: defaultLayouts[ LAYOUT_LIST ].layout,
 };
 
+function useDataViewItemCounts( { postType } ) {
+	const { records, totalItems } = useEntityRecords( 'postType', postType, {
+		per_page: -1,
+		status: [ 'any', 'trash' ],
+	} );
+	return useMemo( () => {
+		if ( ! records ) {
+			return {};
+		}
+		const counts = {
+			all: totalItems,
+			drafts: records.filter( ( record ) => record.status === 'draft' )
+				.length,
+			future: records.filter( ( record ) => record.status === 'future' )
+				.length,
+			pending: records.filter( ( record ) => record.status === 'pending' )
+				.length,
+			private: records.filter( ( record ) => record.status === 'private' )
+				.length,
+			published: records.filter(
+				( record ) => record.status === 'publish'
+			).length,
+			trash: records.filter( ( record ) => record.status === 'trash' )
+				.length,
+		};
+		return counts;
+	}, [ records, totalItems ] );
+}
+
 export function useDefaultViews( { postType } ) {
 	const labels = useSelect(
-		( select ) => {
-			const { getPostType } = select( coreStore );
-			return getPostType( postType )?.labels;
-		},
+		( select ) => select( coreStore ).getPostType( postType )?.labels,
 		[ postType ]
 	);
+	const counts = useDataViewItemCounts( { postType } );
+
 	return useMemo( () => {
 		return [
 			{
 				title: labels?.all_items || __( 'All items' ),
 				slug: 'all',
 				icon: pages,
-				view: DEFAULT_POST_BASE,
+				view: {
+					...DEFAULT_POST_BASE,
+					filters: [
+						{
+							field: 'status',
+							operator: OPERATOR_IS_NONE,
+							value: 'trash',
```

Comment:
I pulled this filter across from the original PR, but it effectively does nothing when you disable the filter in the view.

I think we should remove it.

The [query arg builder](https://github.com/WordPress/gutenberg/blob/17c1206bc169b9a186507fef6608d53e4aa7f602/packages/edit-site/src/components/post-list/index.js#L207-L207) in `<PostList />` doesn't seem to take excluded statuses into account.

As mentioned by @Souptik2001  in https://github.com/WordPress/gutenberg/pull/62028, I think the [default statuses should include "trash"](https://github.com/WordPress/gutenberg/blob/79ff3873ed8c1a2d1af9b7f9bc25416023adc801/packages/edit-site/src/components/post-list/index.js#L174-L174) because the "All pages" count includes the trashed items count but doesn't display the items themselves. 

Either that, or we don't add trashed items to the count and do a separate query for "trashed" items.

Regardless, there seems to be a bug when activating/deactivating fields. No records are returned and therefore [getItemId](https://github.com/WordPress/gutenberg/blob/6a60081909d1dfbbb608e7b069dec3eb6881e37e/packages/edit-site/src/components/post-list/index.js#L176-L176) triggers an error

https://github.com/WordPress/gutenberg/blob/6a60081909d1dfbbb608e7b069dec3eb6881e37e/packages/edit-site/src/components/post-list/index.js#L244-L249

Reply:
> No records are returned and therefore [getItemId](https://github.com/WordPress/gutenberg/blob/6a60081909d1dfbbb608e7b069dec3eb6881e37e/packages/edit-site/src/components/post-list/index.js#L176-L176) triggers an error

PR to fix that I think:

https://github.com/WordPress/gutenberg/pull/65227

Reply:
> disable the filter in the view.

Ideally this should not be possible. See https://github.com/WordPress/gutenberg/issues/60468.

If possible I don't think the "All pages" view should include trash; that's not how it works in wp-admin.

Reply:
Okay, I left it as it is (All pages do not include"trashed" items) but removed the filter.